File: services/payment-service/app/Models/OutboxMessage.php
<?php
namespace App\Models;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Concerns\HasUuids;
class OutboxMessage extends Model {
    use HasUuids;
    protected $table = 'outbox_messages';
    public $timestamps = false;
    protected $fillable = ['event_type','payload','retry_count','created_at'];
    protected $casts = ['processed_at'=>'datetime','created_at'=>'datetime'];
    public function scopeUnprocessed($query) { return $query->whereNull('processed_at')->where('retry_count','<',3); }
}

File: services/order-service/src/OrderService/Domain/OutboxMessage.cs
namespace OrderService.Domain;

public class OutboxMessage
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid OrderId { get; set; }
    public string EventType { get; set; } = string.Empty;
    public string Payload { get; set; } = string.Empty;
    public DateTime? ProcessedAt { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public int RetryCount { get; set; } = 0;

    public Order Order { get; set; } = null!;
}
